openapi.yaml

```diff
@@ -0,0 +1,2094 @@
+          required: false
+
+    CollectionInfo:
+      type: object
+      properties:
+        total_entries:
+          type: integer
+
+    # Enums
+    TaskState:
+      type: string
+      enum:
+        - success
+        - running
+        - failed
+        - upstream_failed
+        - skipped
+        - up_for_retry
+        - up_for_reschedule
+        - queued
+        - none
+        - scheduled
+
+    DagState:
+      type: string
+      enum:
+        - sucess
+        - running
+        - failed
+
+    TriggerRule:
+      type: string
+      enum:
+        - all_success
+        - all_failed
+        - all_done
+        - one_success
+        - one_failed
+        - none_failed
+        - none_skipped
+        - dummy
+
+    WeightRule:
+      type: string
+      enum:
+        - downstream
+        - upstream
+        - absolute
+
+  # Reusable path, query, header and cookie parameters
+  parameters:
+    # Pagination parameters
+    PageOffset:
+      in: query
+      name: offset
+      required: false
+      schema:
+        type: integer
+        minimum: 0
+      description: The number of items to skip before starting to collect the result set.
+
+    PageLimit:
+      in: query
+      name: limit
+      required: false
+      schema:
+        type: integer
+        minimum: 1
+        default: 100
+      description: The numbers of items to return.
+
+    # Database entity fields
+    ConnectionID:
+      in: path
+      name: connection_id
+      schema:
+        type: integer
+      required: true
+      description: The connection ID
+
+    DAGID:
+      in: path
+      name: dag_id
+      schema:
+        type: integer
+      required: true
+      description: The DAG ID
+
+    DAGRunID:
+      in: path
+      name: dag_run_id
+      schema:
+        type: integer
+      required: true
+      description: The DAG Run ID
+
+    TaskID:
+      in: path
+      name: dag_id
+      schema:
+        type: integer
+      required: true
+      description: The Task ID
+
+    TaskTryNumber:
+      in: path
+      name: task_try_number
+      schema:
+        type: integer
+      required: true
+      description: The Task Try Number
+
+    EventLogID:
+      in: path
+      name: event_log_id
+      schema:
+        type: integer
+      required: true
+      description: The Event Log ID
+
+    ImportErrorID:
+      in: path
+      name: import_error_id
+      schema:
+        type: integer
+      required: true
+      description: The Event Log ID
+
+    PoolID:
+      in: path
+      name: pool_id
+      schema:
+        type: integer
+      required: true
+      description: The Pool ID
+
+    TaskRescheduleID:
+      in: path
+      name: task_fail_id
+      schema:
+        type: integer
+      required: true
+      description: The Task Fail ID
+
+    VariableID:
+      in: query
+      name: variable_id
+      schema:
+        type: string
+      required: true
+      description: The Variable ID
+
+    VariableKey:
```

Comment:
Why not path parameter?

Reply:
It may contain special characters including "/". It will be clearer as the query parameter. This endpoint is very similar to: https://cloud.google.com/data-catalog/docs/reference/rest/v1beta1/entries/lookup